### PR TITLE
Workaround for Nest SDP answer bugs to pass firefox validation

### PR DESCRIPTION
**Pull request by jonnynch**
The change is applied to offer sdp with "mozilla" only, which is Firefox.

1. The smartdevicemanagement google api returns answer sdp with a=sendrecv even offer sdp is a=recvonly
Firefox does not accept the incorrect direction
Workaround: to replace the answer sdp direction to sendonly from sendrecv

2. The smartdevicemanagement google api returns ice candidates without foundation field
Firefox cannot parse the candidate without foundation field
Workaround: assign the foundation field into the ice candidate

Ref: https://github.com/home-assistant/core/issues/103924#issuecomment-2333173718

### CODE REVIEW

**Comment by allenporter** [comment]
Looks good, just some lint errors to resolve

**Comment by jonnynch** [comment]
updated
it would be good if the mypy command can be included in pre-commit hook or README.md as well

**Comment by allenporter** [comment]
> updated it would be good if the mypy command can be included in pre-commit hook or README.md as well

It's there https://github.com/allenporter/python-google-nest-sdm/blob/d1701886af112cca971c74733b0e663d625142a9/.pre-commit-config.yaml#L21 -- it's not working?

**Comment by jonnynch** [comment]
Maybe because I am using wsl in windows
image

Anyway, when I run the command directly, it is working fine
`mypy --install-types --non-interactive --no-warn-unused-ignores .`

File: tests/test_webrtc_util.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
 )
 
 
-def test_fix_mozilla_sdp_answer():
+def test_fix_mozilla_sdp_answer() -> None:
     """
     Test the fix in the SDP for Firefox.
     """
```
Comment (allenporter) [inline tests/test_webrtc_util.py:16]:
```suggestion
    """Test the fix in the SDP for Firefox."""
```

File: tests/test_webrtc_util.py [hunk 2]
```diff
@@ -86,7 +86,7 @@ def test_fix_mozilla_sdp_answer():
     assert fixed_sdp == answer_sdp
 
 
-def test_get_media_direction():
+def test_get_media_direction() -> None:
     """
     Test getting the direction in the SDP.
     """
```
Comment (allenporter) [inline tests/test_webrtc_util.py:92]:
```suggestion
    """Test getting the direction in the SDP."""
```

File: tests/test_webrtc_util.py [hunk 3]
```diff
@@ -112,7 +112,7 @@ def test_get_media_direction():
     assert direction is None
 
 
-def test_update_direction_in_answer():
+def test_update_direction_in_answer() -> None:
     """
     Test updating the direction in the SDP answer.
     """
```
Comment (allenporter) [inline tests/test_webrtc_util.py:118]:
```suggestion
def test_update_direction_in_answer() -> None:
    """Test updating the direction in the SDP answer."""
```

File: tests/test_webrtc_util.py [hunk 4]
```diff
@@ -152,7 +152,7 @@ def test_update_direction_in_answer():
     assert new_sdp == expected_sdp
 
 
-def test_add_foundation_to_candidates():
+def test_add_foundation_to_candidates() -> None:
     """
     Test adding a foundation value to ICE candidates.
     """
```
Comment (allenporter) [inline tests/test_webrtc_util.py:158]:
```suggestion
def test_add_foundation_to_candidates() -> None:
    """Test adding a foundation value to ICE candidates."""
```

**Comment by allenporter** [comment]
I've released 5.0.1 that contains the fixes. Are you up for sending the version bump to the home assistant nest integration? I think that's the last thing to get your fixes in.

**Comment by allenporter** [comment]
Ok I'm traveling so it may be a few days

**Comment by jonnynch** [comment]
Eventually raised https://github.com/home-assistant/core/pull/125706

**Pull request merged**